crates/hyperswitch_connectors/src/connectors/nomupay.rs

```diff
@@ -0,0 +1,799 @@
+where
+    Self: ConnectorIntegration<Flow, Request, Response>,
+{
+    fn build_headers(
+        &self,
+        req: &RouterData<Flow, Request, Response>,
+        _connectors: &Connectors,
+    ) -> CustomResult<Vec<(String, masking::Maskable<String>)>, errors::ConnectorError> {
+        let mut header = vec![(
+            headers::CONTENT_TYPE.to_string(),
+            self.get_content_type().to_string().into(),
+        )];
+        let mut api_key = self.get_auth_header(&req.connector_auth_type)?;
+        header.append(&mut api_key);
+        Ok(header)
+    }
+}
+
+impl TryFrom<&Option<pii::SecretSerdeValue>> for nomupay::NomupayMetadata {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(meta_data: &Option<pii::SecretSerdeValue>) -> Result<Self, Self::Error> {
+        let metadata: Self = utils::to_connector_meta_from_secret::<Self>(meta_data.clone())
+            .change_context(errors::ConnectorError::InvalidConnectorConfig {
+                config: "metadata",
+            })?;
+        Ok(metadata)
+    }
+}
+
+/*
+fn get_body_in_map_form(body: &RequestContent)->Result<Map<String, Value>, errors::ConnectorError> {
+    let error = errors::ConnectorError::GenericError { error_message: "josh error".to_string(), error_object: Value::String("unable to create the body".to_string())};
+    match body {
+        RequestContent::Json(json_body) => {
+            let json_str;
+            if serde_json::to_string(&json_body).is_err(){
+                return Err(error)
+            }
+            else{
+                json_str = serde_json::to_string(&json_body).unwrap();
+            }
+
+            // Parse the JSON string into `serde_json::Value`
+            let value: Value ;
+            if serde_json::from_str::<String>(&json_str).is_err(){
+                return Err(error)
+            }
+            else{
+                value=serde_json::from_str(&json_str).unwrap();
+            }
+
+            // Ensure the parsed JSON is an object and convert it to `Map<String, Value>`
+            if let Value::Object(map) = value {
+                // Now you have a `Map<String, Value>`
+                println!("Parsed Map: {:?}", map);
+                Ok(map)
+            } else {
+                return Err(error)
+            }
+        },
+        _=> return Err(error)
+    }
+}
+*/
+
+fn get_private_key(
+    metadata: Option<pii::SecretSerdeValue>,
+) -> Result<String, errors::ConnectorError> {
+    match nomupay::NomupayMetadata::try_from(&metadata) {
+        Ok(nomupay_metadata) => Ok(nomupay_metadata.private_key),
+        Err(_e) => Err(errors::ConnectorError::NoConnectorMetaData),
+    }
+}
+
+fn box_to_jwt_payload(
+    body: Box<dyn masking::ErasedMaskSerialize + Send>,
+) -> Result<JwtPayload, errors::ConnectorError> {
+    let error = errors::ConnectorError::GenericError {
+        error_message: "josh error".to_string(),
```

Comment:
What is josh error? Can we return the actual error in case an error occurs during the steps?